### PR TITLE
Add method to update role permissions

### PR DESCRIPTION
**Pull request by vivangkumar**
I've never written Typescript before, hopefully most of it is right..

### CODE REVIEW

File: src/chatkit.ts [hunk 1]
```diff
@@ -367,6 +367,36 @@ export default class Chatkit {
     })
   }
 
+  updatePermissionsForGlobalRole(
+    roleName: string,
+    permissionsToAdd: Array<string>,
+    permissionsToRemove: Array<string>,
+  ): Promise<any> {
+    var permsToCheck: Array<string> = permissionsToadd.concat(permissionsToRemove)
```
Comment (hamchapman) [inline src/chatkit.ts]: permissionsToadd -> permissionsToAdd

File: src/chatkit.ts [hunk 2]
```diff
@@ -367,6 +367,36 @@ export default class Chatkit {
...
+    (permsToCheck).forEach((perm) => {
```
Comment (hamchapman) [inline src/chatkit.ts]: Don't need brackets around `permsToCheck`

File: src/chatkit.ts [hunk 3]
```diff
@@ -367,6 +367,36 @@ export default class Chatkit {
...
+      if (validGlobalPermissions.indexOf(perm) < 0) {
```
Comment (hamchapman) [inline src/chatkit.ts]: `perm: string` to make the compiler happy

File: src/chatkit.ts [hunk 4]
```diff
@@ -367,6 +367,36 @@ export default class Chatkit {
...
+        throw new Error(`Permission value "${perm}" is invalid`);
+      }
+    })
+
+    return updatePermissionsForRole(roleName, 'global', permissionsToadd, permissionsToRemove)
```
Comment (hamchapman) [inline src/chatkit.ts]: `updatePermissionsForRole` -> `this.updatePermissionsForRole`

`permissionsToadd` -> `permissionsToAdd`

File: src/chatkit.ts [hunk 5]
```diff
@@ -367,6 +367,36 @@ export default class Chatkit {
...
+  }
+
+  updatePermissionsForRoomRole(
+    roleName: string,
+    permissionsToAdd: Array<string>,
+    permissionsToRemove: Array<string>,
+  ): Promise<any> {
+    var permsToCheck: Array<string> = permissionsToadd.concat(permissionsToRemove)
```
Comment (hamchapman) [inline src/chatkit.ts]: `permissionsToadd` -> `permissionsToAdd`

File: src/chatkit.ts [hunk 6]
```diff
@@ -367,6 +367,36 @@ export default class Chatkit {
...
+    permsToCheck.forEach((perm) => {
```
Comment (hamchapman) [inline src/chatkit.ts]: `perm: string`

File: src/chatkit.ts [hunk 7]
```diff
@@ -367,6 +367,36 @@ export default class Chatkit {
...
+      if (validRoomPermissions.indexOf(perm) < 0) {
+        throw new Error(`Permission value "${perm}" is invalid`);
+      }
+    })
+
+    return updatePermissionsForRole(roleName, 'room', permissionsToadd, permissionsToRemove)
```
Comment (hamchapman) [inline src/chatkit.ts]: `updatePermissionsForRole` -> `this.updatePermissionsForRole`

`permissionsToadd` -> `permissionsToAdd`

File: src/chatkit.ts [hunk 8]
```diff
@@ -394,6 +424,36 @@ export default class Chatkit {
...
+    scope: string,
+    permissionsToadd: Array<string>,
+    permissionsToRemove: Array<string>,
+  ): Promise<any> {
+    if (len(permissionsToadd) == 0 && len(permissionsToRemove) == 0) {
+      throw new Error(`Either permissionsToAdd or permissionsToRemove is required`)
+    }
+
+    var body: object = {}
+    if (len(permissionsToadd) > 0) {
```
Comment (hamchapman) [inline src/chatkit.ts]: Not Go 😛

`permissionsToAdd.length`

`permissionsToadd` -> `permissionsToAdd`

File: src/chatkit.ts [hunk 9]
```diff
@@ -394,6 +424,36 @@ export default class Chatkit {
     });
   }
 
+  private updatePermissionsForRole(
+    roleName: string,
+    scope: string,
+    permissionsToadd: Array<string>,
+    permissionsToRemove: Array<string>,
+  ): Promise<any> {
+    if (len(permissionsToadd) == 0 && len(permissionsToRemove) == 0) {
```
Comment (hamchapman) [inline src/chatkit.ts]: `len` -> `length`

`permissionsToadd` -> `permissionsToAdd`

`==` -> `===`

File: src/chatkit.ts [hunk 10]
```diff
@@ -394,6 +424,36 @@ export default class Chatkit {
...
+      throw new Error(`Either permissionsToAdd or permissionsToRemove is required`)
+    }
+
+    var body: object = {}
+    if (len(permissionsToadd) > 0) {
+      body['add_permissions'] = permissionsToadd
```
Comment (hamchapman) [inline src/chatkit.ts]: `permissionsToadd` -> `permissionsToAdd`

File: src/chatkit.ts [hunk 11]
```diff
@@ -394,6 +424,36 @@ export default class Chatkit {
...
+    }
+
+    if (len(permissionsToRemove) > 0) {
```
Comment (hamchapman) [inline src/chatkit.ts]: `len` -> `length`

File: src/chatkit.ts [hunk 12]
```diff
@@ -394,6 +424,36 @@ export default class Chatkit {
...
+    roleName: string,
+    scope: string,
+    permissionsToadd: Array<string>,
+    permissionsToRemove: Array<string>,
+  ): Promise<any> {
+    if (len(permissionsToadd) == 0 && len(permissionsToRemove) == 0) {
+      throw new Error(`Either permissionsToAdd or permissionsToRemove is required`)
+    }
+
+    var body: object = {}
```
Comment (hamchapman) [inline src/chatkit.ts]: Don't type as `object`, type as `any` or create an interface like at the top of the file, e.g.:

```typescript
export interface GetRoomMessagesOptions {
  initial_id?: string;
  direction?: string;
  limit?: number;
}
```

File: src/chatkit.ts (hunk 9 above)
Comment (vivangkumar) [reply src/chatkit.ts]: Completely forgot I'm writing Javascript...

File: src/chatkit.ts (hunk 8 above)
Comment (vivangkumar) [reply src/chatkit.ts]: Been writing Go for far too long..

**Comment by vivangkumar** [comment]
@hamchapman should be all fixed now.

**Pull request merged**